src/_guides/language/async-await.md

```diff
@@ -0,0 +1,344 @@
+}
+
+Future<String> getUserOrder() {
+  return Future.delayed(Duration(seconds: 4), () => 'Large Latte'); 
+}
+
+main() async {
+  await createOrderMessage();
+}
+
+// Output: 
+// Awaiting user order...
+// (4 second pause)
+// Your order is: Large Latte
+```
+
+Notice that the timing of the output shifts if we move the print statement "Awaiting user order" to the line after the first await keyword in `createOrderMessage`:
+
+```
+void createOrderMessage () async {
+  var order = await getUserOrder();
+  print('Awaiting user order...');
+  print('Your order is: $order');
+}
+
+Future<String> getUserOrder() {
+  return Future.delayed(Duration(seconds: 4), () => 'Large Latte'); 
+}
+
+main() async {
+  await createOrderMessage();
+}
+
+// Output: 
+// (4 seconds pause)
+// Awaiting user order...
+// Your order is: Large Latte
+```
+
+
+## Practice Using async and await
+
+
+
+The exercise below is created with partially completed code snippets as a failing unit test, and you will be writing code and make it pass. Verify your code by clicking the Run button. 
+
+Add the following:
+
+* Part 1: Add logic to the `reportUserRole` function so that it returns a Future that produces the String result ‘User role: <user role>!'. The <user role> can be obtained by calling the provided async function `getRole` which returns a `String` describing the user role. Example return value from `reportUserRole()`: `'User role: tester'`
+* Part 2: Write an `async` function `reportLogins` that returns the string: ‘Total number of logins: <login amount>'. The <login amount> can be obtained by calling the provided asynchronous function `getLoginAmount`, which returns an `int` representing the number of times that the user has logged in. Example return value from `reportLogins(): 'Total number of logins: 57'`
+
+
+## Handling errors
+
+
+
+`Async` functions can handle errors using try-catch:
+
+```
+void createOrderMessage () async {
+  try {
+    var order = await getUserOrder();
+    print('Awaiting user order...');
+  } catch (err) {
+    print('Caught error: $err');
+  }
+}
+
+Future<String> getUserOrder() {
+  // Imagine that this function is more complex and throws an exception
+  var str = Future.delayed(Duration(seconds: 4), () => throw 'Cannot locate user order');
+  return str;
+}
+
+main() async {
+  await createOrderMessage();
+}
+
+// Caught error: Cannot locate user order
+```
+
+The try-catch code behaves in the same way with asynchronous code as it does with synchronous code: if the code within the `try` block throws an exception, the code inside the `catch` clause executes. 
+
+
+## Practice Handling Errors with async and awaitImplement code to accomplish the following:
+
+
+
+* Make `changeUsername` call the provided asynchronous function `getNewUsername`, and return its result
+* `changeUsername` must catch and return any errors thrown by getNewUsername, prefixing the returned error with the following string: `Error:` 
```

Comment:
=> The `changeUsername` method must catch and return any errors thrown by `getNewUsername`.
      Prefix the returned error with the string `Error:`.